test/models/trip_test.dart

```diff
@@ -0,0 +1,369 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:motis_mitfahr_app/rides/models/ride.dart';
+import 'package:motis_mitfahr_app/util/search/position.dart';
+import 'package:motis_mitfahr_app/util/trip/trip.dart';
+
+import '../util/factories/drive_factory.dart';
+import '../util/factories/ride_factory.dart';
+import '../util/factories/trip_factory.dart';
+import '../util/mock_server.dart';
+import '../util/mock_server.mocks.dart';
+
+void main() {
+  final MockUrlProcessor tripProcessor = MockUrlProcessor();
+  //setup muss in jeder Testklasse einmal aufgerufen werden
+  setUp(() async {
+    MockServer.setProcessor(tripProcessor);
+  });
+
+  Object randomTripFactory() {
```

Comment:
the return type could be `TripFactory`, then you would not need the casting every time

Reply:
It does not see RideFactory or RideFactory as an instance of TripFactory when I declare TripFactory as a Return type. The reason could be since they don't have a constructor.